src/arkode/arkode_io.c

```diff
@@ -588,6 +611,35 @@ int arkSetFixedStep(void *arkode_mem, realtype hfixed)
   }
   ark_mem = (ARKodeMem) arkode_mem;
 
+  /* Remove current SUNControl object */
+  retval = SUNControlSpace(ark_mem->hcontroller, &lenrw, &leniw);
+  if (retval == SUNCONTROL_SUCCESS) {
+    ark_mem->liw -= leniw;
+    ark_mem->lrw -= lenrw;
+  }
+  SUNControlDestroy(ark_mem->hcontroller);
+  ark_mem->hcontroller = NULL;
```

Comment:
Is it necessary to destroy the controller here or should this only happen if `hfixed == 0` and a controller does not already exist?

Reply:
This call shouldn't hurt anything, since (a) if the controller did not exist then `Destroy` will immediately return with success, and (b) if some controller did already exist and a user called this with `hfixed == 0` then this should indicate a return to the ARKODE "default" PID controller.  

I suppose that this could cause unexpected results in the [very odd] corner case where a user sets up a custom controller, attaches it to ARKODE, and then calls `SetFixedStep` with `hfixed==0`.  Do you actually think that we need to support this kind of usage?